format.go

```diff
@@ -0,0 +1,122 @@
+package lorg
+
+import "strings"
+
+// Format is the actual Formatter which used by Log structure for formatting
+// log records before writing log records into Log.output.
+//
+// Do not instantiate Format instance without using NewFormat.
+type Format struct {
+	formatting   string
+	compiled     bool
+	replacements []replacement
+	placeholders map[string]Placeholder
+}
+
+type replacement struct {
+	value            string
+	placeholder      Placeholder
+	placeholderValue string
+}
+
+// NewFormat creates Format instance with specified formatting and default
+// placeholders: level (PlaceholderLevel), date (PlaceholderDate), line
+// (PlaceholderLine) and file (PlaceholderFile).
+//
+// Format placeholders can be changed or added using SetPlaceholders or
+// SetPlaceholder methods.
+func NewFormat(formatting string) *Format {
+	format := &Format{
+		formatting: formatting,
+	}
+
+	// we are should not assing format.placeholders to defaultPlaceholders
+	// because maps in go passes by reference.
+
+	format.SetPlaceholders(defaultPlaceholders)
+
+	return format
+}
+
+// SetPlaceholder sets specified placeholder with specified placeholder name
+// for given format.
+func (format *Format) SetPlaceholder(name string, placeholder Placeholder) {
+	format.Reset()
+
+	format.placeholders[name] = placeholder
+}
+
+// SetPlaceholders sets specified placeholders for given format.
+func (format *Format) SetPlaceholders(placeholders map[string]Placeholder) {
+	format.Reset()
+
+	format.placeholders = map[string]Placeholder{}
```

Comment:
This method should replace all placeholders that allready set? If not, then this assigment should be in `if` block (`if format.placeholders == nil` for example).